components/style/properties/data.py

```diff
@@ -78,6 +82,12 @@ def __init__(self, style_struct, name, animatable=None, derived_from=None, keywo
         self.depend_on_viewport_size = depend_on_viewport_size
         self.derived_from = (derived_from or "").split()
         self.complex_color = complex_color
+        self.cast_type = cast_type
+
+        # This is a workaround for gecko constants with unexpected names i.e., a constant
+        # corresponding to a keyword 'foo-bar' declared as FOOBAR instead of FOO_BAR
+        if self.keyword is not None:
+            self.keyword.ignore_hyphens = ignore_hyphens
```

Comment:
I think this feels a lot like a hack. If we want to both work-around it and auto-generate it, I think instead of this we should be passing a "keyword to constant" map instead, and then look it up in `gecko_constant` or replace otherwise.

What do you think?